pkg/endpointstate/restorer.go

```diff
@@ -3,15 +3,12 @@
 
 package endpointstate
 
-import "context"
-
-// Restorer wraps a method to wait for endpoints restoration.
-type Restorer interface {
-	// WaitForEndpointRestore blocks the caller until either the context is
-	// cancelled or all the endpoints have been restored from a previous run.
-	WaitForEndpointRestore(ctx context.Context) error
-
-	// WaitForInitialPolicy blocks the caller until either the context is
-	// cancelled or initial policies of all restored endpoints have been computed.
-	WaitForInitialPolicy(ctx context.Context) error
-}
+// EndpointsRestored is an empty type used for promise.Promise.
+// Waiting on that promise blocks the caller until either the context is
+// cancelled or all the endpoints have been restored from a previous run.
```

Comment:
Should also mention that this is closed when all endpoints are *regenerated*. Sometimes we use "restored" to indicate just reading the checkpointed state.